src/otx/algo/detection/heads/yolo_head.py

```diff
@@ -0,0 +1,729 @@
+                                },
+                            ),
+                        )
+                        self.module.append(
+                            Conv2dModule(
+                                conv_channels[1][0],
+                                conv_channels[1][1],
+                                3,
+                                stride=2,
+                                padding=auto_pad(3),
+                                normalization=nn.BatchNorm2d(conv_channels[1][1], eps=1e-3, momentum=3e-2),
+                                activation=nn.SiLU(inplace=True),
+                            ),
+                        )
+                        self.module.append(RepNCSPELAN(csp_channel[0], csp_channel[1], part_channels=csp_channel[2]))
+                    else:
+                        self.module.append(
+                            aux_aconv_adown_object(aux_aconv_adown_channel[0], aux_aconv_adown_channel[1]),
+                        )
+                        self.module.append(
+                            set_info_into_instance({"module": CBFuse(cbfuse_index), "source": cbfuse_source}),
+                        )
+                        self.module.append(
+                            set_info_into_instance(
+                                {
+                                    "module": RepNCSPELAN(csp_channel[0], csp_channel[1], part_channels=csp_channel[2]),
+                                    "tags": f"A{idx+2}",
+                                },
+                            ),
+                        )
+                        aux_output_channels.append(csp_channel[1])
+
+            self.module.append(
+                set_info_into_instance(
+                    {
+                        "module": MultiheadDetection(aux_output_channels, num_classes),
+                        "source": ["A3", "A4", "A5"],
+                        "tags": "AUX",
+                        "output": True,
+                    },
+                ),
+            )
+
+    @property
+    def is_aux(self) -> bool:
+        """Check if the head has an auxiliary head."""
+        return bool(self.aux_cfg)
+
+    def forward(self, outputs: dict[int | str, Tensor], *args, **kwargs) -> tuple[Tensor, None] | tuple[Tensor, Tensor]:
+        """Forward function."""
+        for layer in self.module:
+            if hasattr(layer, "source") and isinstance(layer.source, list):
+                model_input = [outputs[idx] for idx in layer.source]
+            else:
+                model_input = outputs[getattr(layer, "source", -1)]  # type: ignore[arg-type]
+            x = layer(model_input)
+            outputs[-1] = x
+            if hasattr(layer, "tags"):
+                outputs[layer.tags] = x
+
+        if self.is_aux:
+            return outputs["Main"], outputs["AUX"]
+        return outputs["Main"], None
+
+    def loss(self, x: tuple[Tensor], entity: DetBatchDataEntity) -> dict:
```

Comment:
Could you align it with my PR? Or it is better to do after those PR merged, of course

Reply:
Sure, I'll update it after merging your iseg refactoring PR.